File: LISTA 07 - EXTRA (PROLOG)/wc.pl
%	LINGUAGEM DE PROGRAMAÇÃO 2017/ 1
%	PROFESSOR: Carlos Camarão
%	ALUNO: Rômulo Rafael da Silva

%	LISTA 1 - Ex. 4

countchar([ ],0).
countchar([_| Cauda], N) :-
countchar(Cauda, N1),
N is N1 + 1.

countword(Elem,[ ],0).
countword(Elem,[Elem|Y],N) :-
countword(Elem,Y,N1),
N is N1 + 1.
countword(Elem,[Elem1|Y], N) :-
Elem \== Elem1,
countword(Elem,Y,N)
N is N1 + 1.

countline(Elem,[ ],0).
countline(Elem,[Elem|Y],N) :-
countline(Elem,Y,N1),
N is N1 + 1.
countline(Elem,[Elem1|Y], N) :-
Elem \== Elem1,
countline(Elem,Y,N)
N is N1 + 1.

wc(X) :-
	write('('), countchar(X), write(' '),
	countword(' ',X), write(' '), countline('\l', X).

File: LISTA 06 - PYTHON/wc.py
#	LINGUAGEM DE PROGRAMAÇÃO 2017/1
#	PROFESSOR: Carlos Camarão
#	ALUNO: Rômulo Rafael da Silva - 2012055308

# Implementação 1: input arquivo texto

def fileCount(filename):
    txt=open(filename).read()
    countline=len(txt.split("\n"))
    countword=len(txt.split())
    countchar=len(txt.replace("\n","")) # para não contar o '\n'
    return (countchar, countword, countline)	

# Implementação 2: input padrão

def wc (texto):
	countchar = 0
	countword = 0
	countline = 0
	ehstring = False
	naochar = False
	for n in texto:
		countchar += 1
		if (not(n.isspace())):
			ehstring = True
		if (n == "\n" and ehstring):
			countword += 1
			naochar = True
			ehstring = False
		elif (n.isspace() and ehstring):
			countword += 1
			ehstring = False
		elif (n == "\n"):
			naochar = True
			countline += 1
		elif (naochar):
			naochar = False
			countline += 1
	if (ehstring):
		countword += 1
	return (countchar, countword, countline)